lib/internal/http2/core.js

```diff
@@ -1471,6 +1471,11 @@ class Http2Stream extends Duplex {
     this.once('finish', onHandleFinish);
     this.on('resume', streamOnResume);
     this.on('pause', streamOnPause);
+
+    // Set close emitted, so the stream destruction does not
+    // emit them
+    this._readableState.closeEmitted = true;
+    this._writableState.closeEmitted = true;
```

Comment:
I'm not very fond of this lines. A current goal is to reduce (or totally remove) the number of access to `_readableState` and `_writableState`. However, this requires it. Maybe emitting  `'close'` automatically can be moved to an option?

Reply:
This is only because http2 currently emits a 'code' argument with the close event. The idea is to get rid of that in http2 and move the 'code' to a property down the line, but this ensures backwards compat right now.

Should I add a comment explaining that?

Reply:
What I mean is that in both `fs` and `http2` for compat reason we need to not emit `'close'` on destroy. I think this might actually be a legit use case, and so we should support it with an option at construction.

Reply:
The `fs` thing is legit, but we should just get rid of the code in `http2` imo, but that's another discussion. I like the idea of an option, adding it.